src/data-access/repositories/user/knexUserRepository.js

```diff
@@ -0,0 +1,42 @@
+import { db } from '../../db.js';
+
+export const userRepository = {
+    async getSuggested(loginSubstring, limit) {
+        const users = await db('users')
+            .where('isDeleted', false)
+            .whereLike('login', `%${loginSubstring}%`)
+            .orderBy('login')
+            .limit(limit);
+        return users;
+    },
+    async find(id) {
+        const [user] = await db('users')
+            .where({
+                id,
+                'isDeleted': false
+            });
+        return user;
+    },
+    async add(userData) {
+        const [user] = await db('users')
+            .insert(userData)
+            .returning(['id', 'login', 'age']);
```

Comment:
usually I would return all fields from repository and have consistent response from repository/service. If consumer (other layer that uses the repo/service) need only subset of these fields - it's up to him to cut redundant
- repository `add` method should not depend on what fields you have to return in API response. return correct response - api layer responsibility
- other user repository/service consumers might require missing fields